src/main/kotlin/com/openlattice/data/storage/DataDeletionService.kt

```diff
@@ -140,38 +136,94 @@ class DataDeletionService(
 
         val includeClearedEdges = deleteType == DeleteType.Hard
 
-        val entitySetIdToEntityDataKeysMap: Map<UUID, List<EntityDataKey>> = dgm
+        val entitySetIdToEntityDataKeys: Map<UUID, Set<UUID>> = dgm
                 .getEdgesConnectedToEntities(entitySetId, entityKeyIds, includeClearedEdges)
-                .filter { edge -> edge.dst.entitySetId == entitySetId && filteringNeighborEntitySetIds.contains(edge.src.entitySetId) || edge.src.entitySetId == entitySetId && filteringNeighborEntitySetIds.contains(edge.dst.entitySetId) }
+                .filter { edge ->
+                    edge.dst.entitySetId == entitySetId && filteringNeighborEntitySetIds.contains(edge.src.entitySetId)
```

Comment:
this looks like it needs parenthesis, based on how I'm interpreting this logic

Reply:
ie `(a && b) || (c && d)`

Reply:
Even without parenthesis it would interpret it this way since AND > OR, but will add the them to make it clear.